### ISSUE TITLE
feat: optional GFPGAN face enhancement model download

### ISSUE THREAD

**Issue by tadg-paul** [tigger04/superscale#1]
## Problem

GFPGAN is a dedicated face enhancement model that significantly improves upscaled faces (eyes, teeth, skin texture). Real-ESRGAN's Python CLI offers this via `--face_enhance`. Superscale currently excludes it entirely.

## Why it's not bundled

GFPGAN's pretrained weights embed components with non-commercial licences:

- **StyleGAN2** (NVIDIA): non-commercial use only
- **DFDNet**: CC BY-NC-SA 4.0 (non-commercial, share-alike)

GFPGAN's Apache-2.0 licence explicitly carves these out. Bundling the weights would constrain Superscale's own licence and create confusion about what downstream users can legally do.

Note: the licence restrictions apply to **the model weights**, not to images produced by the model. Users can use face-enhanced images commercially.

## Solution

Support GFPGAN as an **optional user-initiated download** that, once installed, runs automatically.

### Behaviour

1. **Download is explicit**: `superscale --download-face-model` with licence notice and confirmation
2. **Once downloaded, face enhancement is automatic**: whenever faces are detected in an upscaled image, they are enhanced — no flag needed
3. **Opt-out available**: `--no-face-enhance` skips face enhancement even when the model is present
4. **Without the model**: face enhancement is silently skipped (no error — user simply hasn't opted in)

### Pipeline ordering

Face enhancement runs **after** upscaling. This is the same order used by the upstream [Real-ESRGAN `--face_enhance` flag](https://github.com/xinntao/Real-ESRGAN):

```
Input → Tile → Upscale (Real-ESRGAN) → Stitch → Face Enhance (GFPGAN) → Output
```

**Rationale:** GFPGAN works best on faces at reasonable resolution (~512×512 face crops). Running it on the upscaled output gives it a much larger face region with more detail to refine than the original low-res input.

### Face detection

Use Apple's built-in [VNDetectFaceRectanglesRequest](https://developer.apple.com/documentation/vision/vndetectfacerectanglesrequest) (available since macOS 10.15) to detect face regions in the upscaled image. This keeps face detection native and fast, and only the enhancement model itself needs to be downloaded.

The face enhancement pass:

1. Detect face bounding boxes in the upscaled image using `VNDetectFaceRectanglesRequest`
2. For each detected face, crop with padding (1.5× the bounding box, matching upstream)
3. Run the cropped face through the GFPGAN CoreML model
4. Blend the enhanced face back into the upscaled image (feathered edges to avoid seams)

### Model download

1. `superscale --download-face-model` fetches the pre-converted CoreML model from our GitHub release
2. Display a clear licence notice before download: the model is for non-commercial use only due to StyleGAN2/DFDNet components
3. Require explicit user confirmation (interactive prompt or `--accept-licence` flag)
4. Store the downloaded model in `~/Library/Application Support/superscale/models/`
5. The GFPGAN weights are never committed to the repo, never included in Homebrew formula, never redistributed by us

### CLI usage

```bash
# Download the face model (one-time, with licence prompt)
superscale --download-face-model

# From now on, faces are automatically enhanced during upscaling
superscale portrait.png

# Skip face enhancement for this run
superscale --no-face-enhance portrait.png
```

## Acceptance Criteria

| ID | AC | Test | Status |
|----|----|----|--------|
| AC1.1 | Given `--download-face-model`, the CLI downloads GFPGAN weights and stores them in `~/Library/Application Support/superscale/models/` | RT-039: Run CLI with `--download-face-model --accept-licence` → assert model file exists at expected path | passing |
| AC1.2 | Given `--download-face-model` in an interactive terminal, the CLI displays the licence notice and requires confirmation before downloading | UT-009: Run `--download-face-model` in a terminal → assert licence notice is displayed and download waits for confirmation | passing |
| AC1.3 | Given the face model is present, face enhancement runs automatically after upscaling on images containing faces | RT-052: Run FaceEnhancer on image with detected faces → assert face region pixels are modified | passing |
| AC1.4 | Given `--no-face-enhance`, face enhancement is skipped even when the model is present | RT-041: Run CLI with `--no-face-enhance` and face model installed → assert no face enhancement occurs | passing |
| AC1.5 | The GFPGAN weights are excluded from git, Homebrew formula, and all distribution artefacts | RT-042: Assert .gitignore excludes GFPGAN patterns, formula has no GFPGAN resource, model not in tracked files | passing |
| AC1.6 | Face detection uses Apple's Vision framework (`VNDetectFaceRectanglesRequest`), not a bundled detector | RT-043: Run face detection on a test image → assert faces are detected using Vision framework | passing |
| AC1.7 | Given the face model is not present, face enhancement is silently skipped (no error) | RT-041: Run pipeline without face model and without `--no-face-enhance` → assert successful upscale with no face enhancement error | passing |
| AC1.8 | The GFPGAN PyTorch model is converted to CoreML and the `.mlpackage` zip is uploaded to the `models-v1` GitHub release | UT-018: Run `superscale --download-face-model --accept-licence` → model downloads, unzips, and installs successfully | passing |

## Dependencies

- Phase 3 (issue #11) — CLI pipeline must be complete ✅
- CoreML conversion of GFPGAN — feasible, prior art exists (john-rocky/CoreML-Models), coremltools conv_transpose fix landed Feb 2025

## Open questions

- ~~Can GFPGAN be converted to CoreML?~~ **Yes** — confirmed feasible, see implementation comment.
- ~~Should the converted CoreML model also be downloaded (from our release) or should we convert on the user's machine?~~ **From our release** — avoids Python dependency at download time.
- ~~What face crop padding ratio gives the best results?~~ **1.5×** — matching upstream Python implementation.

## Phase

Phase 7 of the [implementation plan](https://github.com/tigger04/superscale/blob/master/docs/implementation-plan.md).
